modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -276,18 +276,17 @@ CrackFrontDefinition::orderCrackFrontNodes(std::set<dof_id_type> &nodes)
     //Loop through the set of crack front nodes, and create a node to element map for just the crack front nodes
     //The main reason for creating a second map is that we need to do a sort prior to the set_intersection.
     //The original map contains vectors, and we can't sort them, so we create sets in the local map.
-    std::map<dof_id_type, std::vector<dof_id_type> > & node_to_elem_map = _mesh.nodeToElemMap();
+    const std::map<dof_id_type, std::vector<dof_id_type> > & node_to_elem_map = _mesh.nodeToElemMap();
     std::map<dof_id_type, std::set<dof_id_type> > crack_front_node_to_elem_map;
 
-    for (std::set<dof_id_type>::iterator nit = nodes.begin(); nit != nodes.end(); ++nit )
+    for (auto node_id : nodes)
```

Comment:
We have a [coding convention](http://mooseframework.org/wiki/CodeStandards/) that says to use a reference....  I don't think "I'm always careful" is a great reason to do break this convention, what if the underlying type used in the `std::set` changes?  Also, you would have to convince us the micro-optimization is necessary with profiling...